Lib/codeop.py

```diff
@@ -102,11 +102,20 @@ def _maybe_compile(compiler, source, filename, symbol):
     try:
         if code:
             return code
-        if not code1 and repr(err1) == repr(err2):
+        if not code1 and _is_syntax_error(err1, err2):
             raise err1
     finally:
         err1 = err2 = None
 
+def _is_syntax_error(err1, err2):
+    rep1 = repr(err1)
+    rep2 = repr(err2)
+    if "was never closed" in rep1 and "was never closed" in rep2:
```

Comment:
We could create a new function for this so it feels less hacky. Something like `_is_retroactive_tokenizer_error()`

Reply:
No, I am satisfied with this and think it ready to merge.  'Hacky' was as much a comment on the existing code as the fix.  'For now' meant until a possible rewrite.  A rewrite would be safer after bugs like this are fixed, however done, and new tests are added.